packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -416,44 +372,137 @@ export class ServerlessCluster extends ServerlessClusterBase {
       engine: props.engine.engineType,
       engineVersion: props.engine.engineVersion?.fullVersion,
       engineMode: 'serverless',
       enableHttpEndpoint: Lazy.any({ produce: () => this.enableDataApi }),
-      kmsKeyId: props.storageEncryptionKey?.keyArn,
-      masterUsername: credentials.username,
-      masterUserPassword: credentials.password?.toString(),
       scalingConfiguration: props.scaling ? this.renderScalingConfiguration(props.scaling) : undefined,
       storageEncrypted: true,
-      vpcSecurityGroupIds: securityGroups.map(sg => sg.securityGroupId),
+      vpcSecurityGroupIds: this.securityGroups.map(sg => sg.securityGroupId),
+    };
+
+    this.connections = new ec2.Connections({
+      securityGroups: this.securityGroups,
+      defaultPort: ec2.Port.tcp(Lazy.number({ produce: () => this.clusterEndpoint.port })),
+    });
+  }
+
+  private renderScalingConfiguration(options: ServerlessScalingOptions): CfnDBCluster.ScalingConfigurationProperty {
+    const minCapacity = options.minCapacity;
+    const maxCapacity = options.maxCapacity;
+
+    if (minCapacity && maxCapacity && minCapacity > maxCapacity) {
+      throw new Error('maximum capacity must be greater than or equal to minimum capacity.');
+    }
+
+    const secondsToAutoPause = options.autoPause?.toSeconds();
+    if (secondsToAutoPause && (secondsToAutoPause < 300 || secondsToAutoPause > 86400)) {
+      throw new Error('auto pause time must be between 5 minutes and 1 day.');
+    }
+
+    return {
+      autoPause: (secondsToAutoPause === 0) ? false : true,
+      minCapacity: options.minCapacity,
+      maxCapacity: options.maxCapacity,
+      secondsUntilAutoPause: (secondsToAutoPause === 0) ? undefined : secondsToAutoPause,
+    };
+  }
+}
+
+/**
+ * Properties for a new Aurora Serverless Cluster
+ */
+export interface ServerlessClusterProps extends ServerlessClusterBaseProps {
+  /**
+   * Credentials for the administrative user
+   *
+   * @default - A username of 'admin' and SecretsManager-generated password
+   */
+  readonly credentials?: Credentials;
+
+  /**
+   * The KMS key for storage encryption.
+   *
+   * @default - the default master key will be used for storage encryption
+   */
+  readonly storageEncryptionKey?: kms.IKey;
+}
+
+/**
+ * Create an Aurora Serverless Cluster
+ *
+ * @resource AWS::RDS::DBCluster
+ *
+ */
+export class ServerlessCluster extends ServerlessClusterNew {
+  /**
+   * Import an existing DatabaseCluster from properties
+   */
+  public static fromServerlessClusterAttributes(
+    scope: Construct, id: string, attrs: ServerlessClusterAttributes,
+  ): IServerlessCluster {
+
+    return new ImportedServerlessCluster(scope, id, attrs);
+  }
+
+  public readonly clusterIdentifier: string;
+  public readonly clusterEndpoint: Endpoint;
+  public readonly clusterReadEndpoint: Endpoint;
+
+  public readonly secret?: secretsmanager.ISecret;
+
+  protected enableDataApi?: boolean;
```

Comment:
Can we move this up to `ServerlessClusterNew`?

Reply:
enableDataApi --> done

`readonly` field: see comment above.